src/compiler/types.ts

```diff
@@ -3146,6 +3146,18 @@ namespace ts {
     /* @internal */
     export type AnyImportSyntax = ImportDeclaration | ImportEqualsDeclaration;
 
+    /* @internal */
+    export type AnyImportOrReExport = AnyImportSyntax | ExportDeclaration;
+
+    /* @internal */
+    export type AnyValidImportOrReExport =
+        | (ImportDeclaration | ExportDeclaration) & { moduleSpecifier: StringLiteral }

```

Comment:
This seems to be incorrect from https://github.com/Microsoft/TypeScript/pull/22495/files#diff-08a3cc4f1f9a51dbb468c2810f5229d3R1607 where moduleSpecifier is not present ?

Reply:
It should be, added it there.